src/NLog/Internal/ExceptionHelper.cs

```diff
@@ -41,34 +42,87 @@ namespace NLog.Internal
     /// </summary>
     internal static class ExceptionHelper
     {
+        private const string LoggedKey = "NLog.ExceptionLoggedToInternalLogger";
+
+        /// <summary>
+        /// Mark this exception as logged to the <see cref="InternalLogger"/>.
+        /// </summary>
+        /// <param name="exception"></param>
+        /// <returns></returns>
+        public static void MarkedAsLoggedToInternalLogger(this Exception exception)
```

Comment:
How about renaming this to `MarkAsLoggedToInternalLogger()`? ("Mark" is the verb)

Alternatively, what about `SetLoggedToInternalLogger()`? I think the verb "Set" is more commonly used for methods like this.